osquery/core/init.cpp

```diff
@@ -155,9 +156,13 @@ void initWorkDirectories() {
 void signalHandler(int num) {
   int rc = 0;
 
+  if (num == SIGUSR1) {
```

Comment:
I might quibble over signal choices, something from `setitimer`, like `SIGXCPU`, feels appropriate. But :shrug: seems fine I guess?

Reply:
Since this is a custom behavior that should only happen in a specific situation, I wanted to use a custom signal that isn't used by anything else with a different meaning.
If we had the watchdog communicating via socket to the worker, that would've been a packet instead.